src/core/qgsfieldformatter.cpp

```diff
@@ -21,6 +21,8 @@
 #include "qgsvectordataprovider.h"
 #include "qgsapplication.h"
 
+const QString QgsFieldFormatter::NULL_VALUE = QStringLiteral( "{2839923C-8B7D-419E-B84B-CA2FE9B80EC7}" );
```

Comment:
```suggestion
const QString QgsFieldFormatter::NULL_VALUE = QStringLiteral( "{00000000-0000-0000-0000-000000000000}" );
```

This would be a very intuitive and self-explanatory magic value

Reply:
Ok, I replaced the value. I had simply copied it from the value map field formatter class. 

I put this in the generic class, because I want to use it for all the widget types that have a combobox (classification, enumaration, relation reference, value map and value relation). The value map PR already exists, #61706, but someone would need to reopen it (I now know that I can avoid the automatic closure by commenting). 

Should I rather introduce a new class between this one and the widget's classes that bundles this null value and the updateValue function?

Reply:
Yes, in this case your approach is fine 👍

Reply:
Currently it's copied up, not moved. Is it possible to move it?

Reply:
I have opened a new [PR](https://github.com/qgis/QGIS/pull/62775) for valuemap now, where the `NULL_VALUE` is removed from QgsValueMapFieldFormatter and all the usages changed. Is that enough or should I incorporate the changes to this PR here (it's quite a lot of files), or should I put all the widget types together into one PR and only split up with commits? I did not want to do that because I think it will get **very** big (there are 6 widgets with comboboxes, and I think there will be many "surprises" along the way).